pulpcore/cli/common/__init__.py

```diff
@@ -149,6 +150,18 @@ def _debug_callback(level: int, x: str) -> None:
 main.add_command(debug)
 
 
+@main.command("shell")
+@click.pass_context
+def pulp_shell(ctx: click.Context) -> None:
+    """Activate an interactive shell-mode"""
+    make_click_shell(
+        ctx.parent,
+        prompt="pulp> ",
+        intro="Starting Pulp3 interactive shell...",
+        hist_file=os.path.join(os.path.expanduser("~"), ".cache/pulp/cli-history"),
```

Comment:
I had hoped, click would provide something like this for cache dirs too:
https://click.palletsprojects.com/en/7.x/api/#click.get_app_dir
But it seems not to be the case. So please disregard my former comment. Using that function should allow this project to stay platform independent.